File: api/src/domain/repos/user.repo.ts
import type { CreateUserInput, UpdateUserInput, User } from '../entities';

export interface IUserRepo {
	create(user: CreateUserInput): Promise<User | null>;
	update(user: UpdateUserInput): Promise<User | null>;
	delete(id: string): Promise<void>;
	findById(id: string): Promise<User | null>;
	findByEmail(email: string): Promise<User | null>;
}

File: api/internal/app/ports/user_repo.go
package ports

import (
	"api/internal/domain"
	"context"
)

type UserRepo interface {
	FindByID(id string, context context.Context) (*domain.User, error)
	FindByEmail(email string, context context.Context) (*domain.User, error)
	Create(user *domain.User, context context.Context) error
	Delete(user *domain.User, context context.Context) error
	Update(id string, updates map[string]any, context context.Context) error
}
